flink-tests/src/test/java/org/apache/flink/test/checkpointing/SavepointITCase.java

```diff
@@ -206,6 +206,7 @@ private void testStopWithSavepointForFlip27Source(boolean drain) throws Exceptio
             client.submitJob(jobGraph).get();
 
             BoundedPassThroughOperator.getProgressLatch().await();
+            waitForAllTaskRunning(cluster.getMiniCluster(), jobId, false);
```

Comment:
Here and in other places:
1. Shouldn't the latch above be enough? Why do we need to additionally wait for all tasks running?
2. If we need `waitForAllTaskRunning()`, do we still need the latch?

Reply:
1. Because the progress latch is toggled inside the operator it is not guaranteed that the Sink is already ready.
2. I thought about that but the latch is guaranteed that we handle some data so savepoint won't be empty. I don't really sure that it is important but I decided to leave both of them - the latch is to be sure that we have data, `waitForAllTaskRunning` is  to be sure that all subtasks are running

Reply:
1. Aren't the operators chained in those tests that you have modified? I presume that was the intention in those tests/latches. If they are not, yes I can see this tests could start failing as a result of https://issues.apache.org/jira/browse/FLINK-22379

Reply:
1. They are not chained since parallelism more than 1
2. Yes, I am sure. In fact, I can easily reproduce this fail. It just needs to add the delay in the initialization phase for the Sink.

Reply:
Ok, the issue here is not with the parallelism, but that some of those tests are disabling chaining in one way or another and that's why sink was in independent task and after FLINK-22379 this could cause failures when triggering savepoint.